File: 1534-count-good-triplets/1534-count-good-triplets.js
/**
 * @param {number[]} arr
 * @param {number} a
 * @param {number} b
 * @param {number} c
 * @return {number}
 */
var countGoodTriplets = function(arr, a, b, c) {
  const l = arr.length
  let ans = 0
  for (let i = 0; i < l - 2; i++) {
    const ai = arr[i]
    for (let j = i + 1; j < l - 1; j++) {
      const bj = arr[j]
      if (ai - bj > a || bj - ai > a) continue
      for (let k = j + 1; k < l; k++) {
        const ck = arr[k]
        if (bj - ck >= -b && bj - ck <= b && ai - ck >= -c && ai - ck <= c) ans++
      }
    }
  }
  return ans
};

File: 1534-count-good-triplets/1534-count-good-triplets.go
func absDiff(a, b int) int {
  if a > b {
    return a - b
  } else {
    return b - a
  }
}
func countGoodTriplets(arr []int, a int, b int, c int) int {
  l := len(arr)
  ans := 0
  for i := 0; i < l - 2; i++ {
    vi := arr[i]
    for j := i+1; j < l - 1; j++ {
      vj := arr[j]
      if absDiff(vi, vj) > a {
        continue
      }
      for k:= j+1; k < l; k++ {
        vk := arr[k]
        if absDiff(vj, vk) <= b && absDiff(vi, vk) <= c {
          ans ++
        }
      }
    }
  }
  return ans
}